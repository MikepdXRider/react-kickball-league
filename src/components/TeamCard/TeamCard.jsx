import React from 'react'
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';


export default function TeamCard({teamDataObj}) {
    return (
        <Link to={`/team/${teamDataObj.id}`}>
            <article style={{border: '1px solid black'}}>
                <h3>{teamDataObj.name}</h3>
                <h6>{teamDataObj.created_at}</h6>
                <h5>{`${teamDataObj.city}, ${teamDataObj.state} `}</h5>
            </article>
        </Link>
    )
}

TeamCard.propType = {
    city: PropTypes.string.isRequired,
    created_at: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    state: PropTypes.string.isRequired
}