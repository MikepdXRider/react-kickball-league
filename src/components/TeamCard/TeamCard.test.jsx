import { screen } from "@testing-library/react";
import { render } from "@testing-library/react";
import TeamCard from "./TeamCard.jsx";


const teamDataObj = {
    created_at: 'test-utc',
    id: 1,
    name: 'test-name',
    team_id: 3,
    city: 'test-city',
    state: 'test-state'
}

it('renders TeamCard component', () => {
    const { container } = render(<TeamCard teamDataObj={teamDataObj} /> )
    const teamName = screen.getByText(teamDataObj.name);

    expect(teamName).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})