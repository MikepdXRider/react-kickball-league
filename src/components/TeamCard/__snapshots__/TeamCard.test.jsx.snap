// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TeamCard component 1`] = `
<div>
  <article
    style="border: 1px solid black;"
  >
    <h3>
      test-name
    </h3>
    <h6>
      test-utc
    </h6>
    <h5>
      test-city, test-state 
    </h5>
  </article>
</div>
`;
